Assets/Scripts/EnemyController.cs

```diff
@@ -78,37 +95,29 @@ void MeleeAttack()
 
     void RangedAttack()
     {
-        if (IsWithinCamera())
+        if (IsWithinCamera() && Time.time > nextFire && playerTransform != null)
         {
-            if (Time.time > nextFire)
+            GameObject lol = Instantiate(projectile, transform.position, Quaternion.identity);
+            ProjectileController proj = lol.GetComponent<ProjectileController>();
+
+            if (proj != null)
             {
-                GameObject projGameObject = Instantiate(projectile, transform.position, Quaternion.identity);
-                ProjectileController proj = projGameObject.GetComponent<ProjectileController>();
-
-                if (proj != null)
-                {
-                    proj.tName = ProjectileController.TargetName.Player;
-                    proj.target = player;
-                }
-                nextFire = Time.time + fireRate;
+                proj.target = playerTransform;
+                proj.senderColl = null; // When Enemies have colliders,
+                                        // TODO: Make a variable for Enemy's collider and put it here
```

Comment:
I think you should just add a collider to the Enemy prefab and use it here as part of this pull request.

Reply:
I think I said I would add a collider to that prefab on `premain`. Now, I'm not planning to do that because it would break enemy projectiles without your change with `senderColl`.